test/unittest/logging/CMakeLists.txt

```diff
@@ -39,5 +39,37 @@ if(NOT ((MSVC OR MSVC_IDE) AND EPROSIMA_INSTALLER))
                 )
         endif()
         add_gtest(LogTests SOURCES ${LOGTESTS_SOURCE})
+
+        set(LOGFILETESTS_SOURCE
+            ${PROJECT_SOURCE_DIR}/src/cpp/log/Log.cpp
+            ${PROJECT_SOURCE_DIR}/src/cpp/log/StdoutConsumer.cpp
+            ${PROJECT_SOURCE_DIR}/src/cpp/log/FileConsumer.cpp
+            LogFileTests.cpp)
+
+        # External sources
+        if(TINYXML2_SOURCE_DIR)
+            list(APPEND LOGFILETESTS_SOURCE
+                ${TINYXML2_SOURCE_DIR}/tinyxml2.cpp
+                )
+        endif()
+
+        include_directories(${TINYXML2_INCLUDE_DIR})
+
+        add_executable(LogFileTests ${LOGFILETESTS_SOURCE})
+        target_compile_definitions(LogFileTests PRIVATE FASTRTPS_NO_LIB)
+        target_include_directories(LogFileTests PRIVATE ${GTEST_INCLUDE_DIRS}
+            ${PROJECT_SOURCE_DIR}/include ${PROJECT_BINARY_DIR}/include)
+        target_link_libraries(LogFileTests ${GTEST_LIBRARIES} ${MOCKS} )
+        if(MSVC OR MSVC_IDE)
+            target_link_libraries(LogFileTests ${PRIVACY}
+                iphlpapi Shlwapi
+                )
+        endif()
+        target_link_libraries(LogFileTests ${PRIVACY} fastcdr)
+        if(TINYXML2_LIBRARY)
```

Comment:
Instead of a conditional IF, use a cmake generator expresion inside `target_link_libraries`.